### PR TITLE
merge ugent hardware templates

### PR DESCRIPTION
**Pull request by kwaegema**

### CODE REVIEW

File: hardware/cpu/intel/xeon_e5_2620_3.pan [hunk 1]
```diff
@@ -0,0 +1,8 @@
+structure template hardware/cpu/intel/xeon_e5_2620_3;
```
Comment (jrha) [inline hardware/cpu/intel/xeon_e5_2620_3.pan]: Does this alreay exist as `xeon_e5_2620v3`?

File: hardware/cpu/intel/xeon_e5_2623_3.pan [hunk 2]
```diff
@@ -0,0 +1,8 @@
+structure template hardware/cpu/intel/xeon_e5_2623_3;
```
Comment (jrha) [inline hardware/cpu/intel/xeon_e5_2623_3.pan]: Ditto.

File: hardware/cpu/intel/xeon_e5_2623_3_ht.pan [hunk 3]
```diff
@@ -0,0 +1,8 @@
+structure template hardware/cpu/intel/xeon_e5_2623_3_ht;
```
Comment (jrha) [inline hardware/cpu/intel/xeon_e5_2623_3_ht.pan]: What does this mean?

Comment (jrha) [reply hardware/cpu/intel/xeon_e5_2623_3_ht.pan]: Is this an e5 2623 v3 with hyperthreading enabled? That's a system setting, not a CPU setting.

File: hardware/cpu/intel/xeon_e5_2623_4.pan [hunk 4]
```diff
@@ -0,0 +1,8 @@
+structure template hardware/cpu/intel/xeon_e5_2623_4;
```
Comment (jrha) [inline hardware/cpu/intel/xeon_e5_2623_4.pan]: Same again.

File: hardware/harddisk/ssd.pan [hunk 5]
```diff
@@ -0,0 +1,6 @@
+
+structure template hardware/harddisk/ssd;
```
Comment (jrha) [inline hardware/harddisk/ssd.pan]: This should really be `hardware/harddisk/generic/ssd`

File: hardware/nic/bnx.pan [hunk 6]
```diff
@@ -0,0 +1,7 @@
+structure template hardware/nic/bnx;
```
Comment (jrha) [inline hardware/nic/bnx.pan]: You are probably better of using one of the templates under `hardware/nic/broadcom/`

File: hardware/ib/connectx.pan [hunk 7]
```diff
@@ -0,0 +1,13 @@
+######################################################
+#
+# template hardware/ib/connectx
+#
+######################################################
```
Comment (jrha) [inline hardware/ib/connectx.pan]: Redundant comment block

File: hardware/ib/connectx2.pan [hunk 8]
```diff
@@ -0,0 +1,18 @@
+######################################################
+#
+# template hardware/ib/connectx2
+#
+######################################################
```
Comment (jrha) [inline hardware/ib/connectx2.pan]: Redundant comment block

File: hardware/ib/infinihostIII.pan [hunk 9]
```diff
@@ -0,0 +1,13 @@
+######################################################
+#
+# template hardware/ib/infinihostIII
+#
+######################################################
```
Comment (jrha) [inline hardware/ib/infinihostIII.pan]: Redundant comment block

File: hardware/ib/qle7340.pan [hunk 10]
```diff
@@ -0,0 +1,13 @@
+######################################################
+#
+# template hardware/ib/qle7340
+#
+######################################################
```
Comment (jrha) [inline hardware/ib/qle7340.pan]: Redundant comment block

File: hardware/nic/bnxt_en.pan [hunk 11]
```diff
@@ -0,0 +1,7 @@
+structure template hardware/nic/bnxt_en;
```
Comment (jrha) [inline hardware/nic/bnxt_en.pan]: Should be under `hardware/nic/by_driver` or `hardware/nic/broadcom/`

File: hardware/nic/connectx3-fdr-10gbe.pan [hunk 12]
```diff
@@ -0,0 +1,7 @@
+structure template hardware/nic/connectx3-fdr-10gbe;
```
Comment (jrha) [inline hardware/nic/connectx3-fdr-10gbe.pan]: These should go under `hardware/nic/mellanox`

File: hardware/nic/generic.pan [hunk 13]
```diff
@@ -0,0 +1,10 @@
+structure template hardware/nic/generic;
```
Comment (jrha) [inline hardware/nic/generic.pan]: @jouvin just added this in #117 as `hardware/nic/generic/generic_nic`, we don't need another one!

File: hardware/nic/mellanox/mt27520.pan [hunk 14]
```diff
@@ -6,5 +6,5 @@ structure template hardware/nic/mellanox/mt27520;
 'pxe' = false;
 'boot' = false;
 'media' = 'ethernet';
-'name' = 'Mellanox Technologies MT27520 Family [ConnectX-3 Pro]';
+'name' = 'Mellanox Technologies MT27520 Family (ConnectX-3 Pro)';
```
Comment (jrha) [inline hardware/nic/mellanox/mt27520.pan]: Why change this? What does `lspci` say?

File: hardware/nic/mellanox/mt27710.pan [hunk 15]
```diff
@@ -6,5 +6,5 @@ structure template hardware/nic/mellanox/mt27710;
 'pxe' = false;
 'boot' = false;
 'media' = 'ethernet';
-'name' = 'Mellanox Technologies MT27710 Family [ConnectX-4 Lx]';
+'name' = 'Mellanox Technologies MT27710 Family (ConnectX-4 Lx)';
```
Comment (jrha) [inline hardware/nic/mellanox/mt27710.pan]: Ditto.

File: hardware/nic/qlcnic.pan [hunk 16]
```diff
@@ -0,0 +1,7 @@
+structure template hardware/nic/qlcnic;
```
Comment (jrha) [inline hardware/nic/qlcnic.pan]: `hardware/nic/qlogic/…`

File: hardware/nic/mellanox/mt27520.pan (hunk 14 above)
Comment (kwaegema) [reply hardware/nic/mellanox/mt27520.pan]: i think this is the reason why:
cfg/standard/hardware/nic/mellanox/mt27520.pan:9: Unwanted space in simple expression in square brackets
'name' = 'Mellanox Technologies MT27520 Family [ConnectX-3 Pro]';

But I guess this should be fixed in panlint instead

**Comment by kwaegema** [comment]
@jrha should be better now :smile:

File: hardware/nic/mellanox/mt27520.pan (hunk 14 above)
Comment (jrha) [reply hardware/nic/mellanox/mt27520.pan]: Indeed, that test shouldn't match inside a string. I've opened an issue.

File: hardware/cpu/amd/opteron_6136.pan [hunk 17]
```diff
@@ -0,0 +1,6 @@
+structure template hardware/cpu/amd/opteron_6136;
+
+"vendor" = "AMD";
+"model"  = "AMD Opteron(TM) Opteron 6136 2.4 GHz (quad core)";
+"speed"  = 2400 * MHz;
+"cores" = 2 * 4;
```
Comment (jrha) [inline hardware/cpu/amd/opteron_6136.pan]: So...`8`?

File: hardware/cpu/amd/opteron_6136.pan [hunk 18]
```diff
@@ -0,0 +1,6 @@
+structure template hardware/cpu/amd/opteron_6136;
+
+"vendor" = "AMD";
+"model"  = "AMD Opteron(TM) Opteron 6136 2.4 GHz (quad core)";
```
Comment (jrha) [inline hardware/cpu/amd/opteron_6136.pan]: This is an 8 core chip...

File: hardware/nic/by_driver/bnx.pan [hunk 19]
```diff
@@ -0,0 +1,7 @@
+structure template hardware/nic/by_driver/bnx;
```
Comment (jrha) [inline hardware/nic/by_driver/bnx.pan]: Is the driver `bnx` or `tg3`?

File: hardware/nic/qlogic/generic.pan [hunk 20]
```diff
@@ -0,0 +1,7 @@
+structure template hardware/nic/qlogic/generic;
+
+include 'hardware/nic/generic/generic_nic';
+
+"driver" = "qlcnic";
+"name"   = "QLogic Corp. 82xx";
```
Comment (jrha) [inline hardware/nic/qlogic/generic.pan]: Is this really a generic QLogic card or an `82xx`?

**Review by jrha — APPROVED** [review]

**Pull request merged**